[quarkus-next] Keycloak CI failure: Cannot invoke "io.quarkus.runtime.Application.isStarted()" because "app" is null

### Description:
Integration tests fails due to the following:
```
jakarta.persistence.PersistenceException: [PersistenceUnit: keycloak-default] Unable to build Hibernate SessionFactory
at io.quarkus.hibernate.orm.runtime.boot.FastBootEntityManagerFactoryBuilder.persistenceException(FastBootEntityManagerFactoryBuilder.java:129)
at io.quarkus.hibernate.orm.runtime.boot.FastBootEntityManagerFactoryBuilder.build(FastBootEntityManagerFactoryBuilder.java:89)
at io.quarkus.hibernate.orm.runtime.FastBootHibernatePersistenceProvider.createEntityManagerFactory(FastBootHibernatePersistenceProvider.java:72)
at jakarta.persistence.Persistence.createEntityManagerFactory(Persistence.java:80)
at jakarta.persistence.Persistence.createEntityManagerFactory(Persistence.java:55)
at io.quarkus.hibernate.orm.runtime.JPAConfig$LazyPersistenceUnit.get(JPAConfig.java:154)
at io.quarkus.hibernate.orm.runtime.JPAConfig$1.run(JPAConfig.java:61)
at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: io.quarkus.runtime.configuration.ConfigurationException: Persistence unit 'keycloak-default' was configured to run with a database version of at least '2.3.232', but the actual version is '2.2.0'. Consider upgrading your database. Alternatively, rebuild your application with 'quarkus.datasource.db-version=2.2.0' (but this may disable some features and/or impact performance negatively).
at io.quarkus.hibernate.orm.runtime.service.QuarkusRuntimeInitDialectFactory.checkActualDbVersion(QuarkusRuntimeInitDialectFactory.java:70)
at io.quarkus.hibernate.orm.runtime.observers.QuarkusSessionFactoryObserverForDbVersionCheck.sessionFactoryCreated(QuarkusSessionFactoryObserverForDbVersionCheck.java:15)
at org.hibernate.internal.SessionFactoryObserverChain.sessionFactoryCreated(SessionFactoryObserverChain.java:35)
at org.hibernate.internal.SessionFactoryImpl.<init>(SessionFactoryImpl.java:323)
at io.quarkus.hibernate.orm.runtime.boot.FastBootEntityManagerFactoryBuilder.build(FastBootEntityManagerFactoryBuilder.java:87)
... 6 more
Exception in thread "Shutdown thread" java.lang.NullPointerException: Cannot invoke "io.quarkus.runtime.Application.isStarted()" because "app" is null
at io.quarkus.runtime.ApplicationLifecycleManager$ShutdownHookThread.run(ApplicationLifecycleManager.java:455)
06:59:53,649 ERROR [org.keycloak.testsuite.actions.AppInitiatedActionDeleteCredentialTest] [AppInitiatedActionDeleteCredentialTest] null()
```

The failure seems to be specific to `-Pauth-server-quarkus`.

Full log can be found here: 
https://github.com/keycloak/keycloak/actions/runs/10481101059

### Steps to reproduce:
```
mvn test -Pauth-server-quarkus -Dtest="org.keycloak.testsuite.actions.AppInitiatedActionDeleteCredentialTest#missingActionId" -pl testsuite/integration-arquillian/tests/base
```

### Additional information: 
The issue seems to be caused by https://github.com/quarkusio/quarkus/pull/42502, as the message from the log indicates:
```
Persistence unit 'keycloak-default' was configured to run with a database version of at least '2.3.232', but the actual version is '2.2.0'.
```
Therefore, H2 upgrade seems to be required.

--- a/pom.xml
+++ b/pom.xml
@@ -92,7 +92,7 @@
         <bouncycastle.bctls-fips.version>1.0.19</bouncycastle.bctls-fips.version>
 
         <dom4j.version>2.1.3</dom4j.version>
-        <h2.version>2.2.224</h2.version>
+        <h2.version>2.3.232</h2.version>
         <hibernate-orm.plugin.version>6.2.13.Final</hibernate-orm.plugin.version>
         <hibernate.c3p0.version>6.2.13.Final</hibernate.c3p0.version>
         <infinispan.version>15.0.8.Final</infinispan.version>

